### PR TITLE
AudioEffects documentation enhancements

### PR DESCRIPTION
**Pull request by betalars**
Enhancing documentation and readability of some audio effects that were not yet documented.

I did some heavy guessing with StereoEnhance, so please double check if the things I determined it to do are actually the things it is intended to do.

### CODE REVIEW

File: doc/classes/AudioEffectFilter.xml [hunk 1]
```diff
@@ -14,22 +14,27 @@
 			Threshold frequency for the filter, in Hz.
 		</member>
 		<member name="db" type="int" setter="set_db" getter="get_db" enum="AudioEffectFilter.FilterDB" default="0">
+			Steepnes of the cutoff curve in dB per octave, also known as order of the filter. Higher orders have a more aggressive cutoff.
```
Comment (AThousandShips) [inline doc/classes/AudioEffectFilter.xml:17]:
```suggestion
			Steepens of the cutoff curve in dB per octave, also known as the order of the filter. Higher orders have a more aggressive cutoff.
```

File: servers/audio/effects/audio_effect_stereo_enhance.cpp
Comment (AThousandShips) [inline servers/audio/effects/audio_effect_stereo_enhance.cpp:1]: Please move these changes to a separate PR, we prefer keeping code changes and documentation changes separate unless they are actually related

File: doc/translations/de.po
Comment (AThousandShips) [inline doc/translations/de.po:1]: These files should not be edited directly here, instead you edit them on the [Hosted Weblate](https://hosted.weblate.org/projects/godot-engine/), see the [localization instructions](https://docs.godotengine.org/en/latest/contributing/documentation/editor_and_docs_localization.html) for more info, once this change has been merged

**Review by AThousandShips — DISMISSED** [review]

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 2]
```diff
@@ -11,11 +11,13 @@
...
 		<member name="pan_pullout" type="float" setter="set_pan_pullout" getter="get_pan_pullout" default="1.0">
-			Values greater than 1.0 increase intensity of any panning on audio passing through this effect, whereas values less than 1.0 will decrease the panning intensity. A value of 0.0 will downmix audio to mono.
-		</member>
-		<member name="surround" type="float" setter="set_surround" getter="get_surround" default="0.0">
+			Amplifies the difference between stereo channels, increasing or descreasing existing panning. A value of 0.0 will downmix stereo to mono. Does not affect a mono signal.
 		</member>
 		<member name="time_pullout_ms" type="float" setter="set_time_pullout" getter="get_time_pullout" default="0.0">
+			Widens sound stage trough phase shifting in cunjuncion with [surround]. Just delays the right channel if [surround] is 0.
+		</member>
+		<member name="surround" type="float" setter="set_surround" getter="get_surround" default="0.0">
```
Comment (AThousandShips) [inline doc/classes/AudioEffectStereoEnhance.xml:19]: This should be kept where it was, properties should be sorted alphabetically

**Comment by AThousandShips** [comment]
Your commit seems not to be linked to your GitHub account. See [why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

File: doc/classes/AudioEffectFilter.xml [hunk 3]
```diff
@@ -14,22 +14,27 @@
...
 		</member>
 		<member name="gain" type="float" setter="set_gain" getter="get_gain" default="1.0">
-			Gain amount of the frequencies after the filter.
+			Boosts frequencies at the cutoff.
```
Comment (Mickeon) [inline doc/classes/AudioEffectFilter.xml:20]: I find this to be worse. The subject is a floating-point property, but the description makes it sound like it's a method that does something when called.

Comment (betalars) [reply doc/classes/AudioEffectFilter.xml:20]: Hm, I see what you mean.

File: servers/audio/effects/audio_effect_stereo_enhance.cpp
Comment (betalars) [reply servers/audio/effects/audio_effect_stereo_enhance.cpp:1]: reverted the changes.

File: doc/translations/de.po
Comment (betalars) [reply doc/translations/de.po:1]: reverted these changes as well

File: doc/classes/AudioEffectFilter.xml (hunk 3 above)
Comment (betalars) [reply doc/classes/AudioEffectFilter.xml:20]: Oh wait I see where I went wrong.

Comment (betalars) [reply doc/classes/AudioEffectFilter.xml:20]:
```suggestion
			Gain amount of the frequencies after the filter.
```

File: doc/classes/AudioEffectFilter.xml [hunk 4]
```diff
@@ -14,6 +14,7 @@
 			Threshold frequency for the filter, in Hz.
 		</member>
 		<member name="db" type="int" setter="set_db" getter="get_db" enum="AudioEffectFilter.FilterDB" default="0">
+			Steepens of the cutoff curve in dB per octave, also known as the order of the filter. Higher orders have a more aggressive cutoff.
```
Comment (AThousandShips) [inline doc/classes/AudioEffectFilter.xml:17]:
```suggestion
			Steepness of the cutoff curve in dB per octave, also known as the order of the filter. Higher orders have a more aggressive cutoff.
```
My bad corrected the typo the wrong way

File: doc/classes/AudioEffectStereoEnhance.xml (hunk 2 above)
Comment (betalars) [reply doc/classes/AudioEffectStereoEnhance.xml:19]: thanks

File: doc/classes/AudioEffectFilter.xml (hunk 4 above)
Comment (betalars) [reply doc/classes/AudioEffectFilter.xml:17]: Oh right. this was the kind of thing I tend to mess up so I just did not double-check. Good catch.

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 5]
```diff
@@ -11,11 +11,13 @@
 	</tutorials>
 	<members>
 		<member name="pan_pullout" type="float" setter="set_pan_pullout" getter="get_pan_pullout" default="1.0">
-			Values greater than 1.0 increase intensity of any panning on audio passing through this effect, whereas values less than 1.0 will decrease the panning intensity. A value of 0.0 will downmix audio to mono.
+			Amplifies the difference between stereo channels, increasing or descreasing existing panning. A value of 0.0 will downmix stereo to mono. Does not affect a mono signal.
 		</member>
 		<member name="surround" type="float" setter="set_surround" getter="get_surround" default="0.0">
+			Widens sound stage trough phase shifting in cunjuncion with [param time_pullout_ms]. Just pans sound to the left channel if [param time_pullout_ms] is 0.
```
Comment (AThousandShips) [inline doc/classes/AudioEffectStereoEnhance.xml:17]:
```suggestion
			Widens sound stage through phase shifting in cunjuncion with [param time_pullout_ms]. Just pans sound to the left channel if [param time_pullout_ms] is 0.
```

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 6]
```diff
@@ -11,11 +11,13 @@
...
 		</member>
 		<member name="time_pullout_ms" type="float" setter="set_time_pullout" getter="get_time_pullout" default="0.0">
+			Widens sound stage trough phase shifting in cunjuncion with [param surround]. Just delays the right channel if [param surround] is 0.
```
Comment (AThousandShips) [inline doc/classes/AudioEffectStereoEnhance.xml:20]:
```suggestion
			Widens sound stage through phase shifting in cunjuncion with [param surround]. Just delays the right channel if [param surround] is 0.
```

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 7]
```diff
@@ -11,11 +11,13 @@
 	</tutorials>
 	<members>
 		<member name="pan_pullout" type="float" setter="set_pan_pullout" getter="get_pan_pullout" default="1.0">
-			Values greater than 1.0 increase intensity of any panning on audio passing through this effect, whereas values less than 1.0 will decrease the panning intensity. A value of 0.0 will downmix audio to mono.
+			Amplifies the difference between stereo channels, increasing or descreasing existing panning. A value of 0.0 will downmix stereo to mono. Does not affect a mono signal.
 		</member>
 		<member name="surround" type="float" setter="set_surround" getter="get_surround" default="0.0">
+			Widens sound stage through phase shifting in cunjuncion with [param time_pullout_ms]. Just pans sound to the left channel if [param time_pullout_ms] is 0.
```
Comment (AThousandShips) [inline doc/classes/AudioEffectStereoEnhance.xml:17]:
```suggestion
			Widens sound stage through phase shifting in cunjuncion with [member time_pullout_ms]. Just pans sound to the left channel if [member time_pullout_ms] is 0.
```

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 8]
```diff
@@ -11,11 +11,13 @@
...
 		</member>
 		<member name="time_pullout_ms" type="float" setter="set_time_pullout" getter="get_time_pullout" default="0.0">
+			Widens sound stage through phase shifting in cunjuncion with [param surround]. Just delays the right channel if [param surround] is 0.
```
Comment (AThousandShips) [inline doc/classes/AudioEffectStereoEnhance.xml:20]:
```suggestion
			Widens sound stage through phase shifting in cunjuncion with [member surround]. Just delays the right channel if [member surround] is 0.
```

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 9]
```diff
@@ -11,11 +11,13 @@
 	</tutorials>
 	<members>
 		<member name="pan_pullout" type="float" setter="set_pan_pullout" getter="get_pan_pullout" default="1.0">
-			Values greater than 1.0 increase intensity of any panning on audio passing through this effect, whereas values less than 1.0 will decrease the panning intensity. A value of 0.0 will downmix audio to mono.
+			Amplifies the difference between stereo channels, increasing or descreasing existing panning. A value of 0.0 will downmix stereo to mono. Does not affect a mono signal.
```
Comment (betalars) [inline doc/classes/AudioEffectStereoEnhance.xml:14]:
```suggestion
			Amplifies the difference between stereo channels, increasing or decreasing existing panning. A value of 0.0 will downmix stereo to mono. Does not affect a mono signal.
```

**Comment by betalars** [comment]
BTW: I have now installed a spell checker for my VSCode.

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 10]
```diff
@@ -11,11 +11,13 @@
 	</tutorials>
 	<members>
 		<member name="pan_pullout" type="float" setter="set_pan_pullout" getter="get_pan_pullout" default="1.0">
-			Values greater than 1.0 increase intensity of any panning on audio passing through this effect, whereas values less than 1.0 will decrease the panning intensity. A value of 0.0 will downmix audio to mono.
+			Amplifies the difference between stereo channels, increasing or decreasing existing panning. A value of 0.0 will downmix stereo to mono. Does not affect a mono signal.
```
Comment (adamscott) [inline doc/classes/AudioEffectStereoEnhance.xml:14]: What a wonderful rewrite, job well done.

**Review by adamscott — APPROVED** [review]
Thanks for the definitions. They seem alright!

**Comment by betalars** [comment]
I'm unsure why but some of the typo fixes did get lost in the first squash, but i did another fixup squash and now alll should be good :)

**Review by clayjohn — APPROVED** [review]

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 11]
```diff
@@ -11,11 +11,13 @@
...
 		</member>
 		<member name="surround" type="float" setter="set_surround" getter="get_surround" default="0.0">
+			Widens sound stage through phase shifting in cunjuncion with [member time_pullout_ms]. Just pans sound to the left channel if [member time_pullout_ms] is 0.
 		</member>
 		<member name="time_pullout_ms" type="float" setter="set_time_pullout" getter="get_time_pullout" default="0.0">
+			Widens sound stage through phase shifting in cunjuncion with [member surround]. Just delays the right channel if [member surround] is 0.
```
Comment (RedMser) [inline doc/classes/AudioEffectStereoEnhance.xml:20]:
```suggestion
			Widens sound stage through phase shifting in conjuncion with [member surround]. Just delays the right channel if [member surround] is 0.
```

File: doc/classes/AudioEffectStereoEnhance.xml [hunk 12]
```diff
@@ -11,11 +11,13 @@
 	</tutorials>
 	<members>
 		<member name="pan_pullout" type="float" setter="set_pan_pullout" getter="get_pan_pullout" default="1.0">
-			Values greater than 1.0 increase intensity of any panning on audio passing through this effect, whereas values less than 1.0 will decrease the panning intensity. A value of 0.0 will downmix audio to mono.
+			Amplifies the difference between stereo channels, increasing or decreasing existing panning. A value of 0.0 will downmix stereo to mono. Does not affect a mono signal.
 		</member>
 		<member name="surround" type="float" setter="set_surround" getter="get_surround" default="0.0">
+			Widens sound stage through phase shifting in cunjuncion with [member time_pullout_ms]. Just pans sound to the left channel if [member time_pullout_ms] is 0.
```
Comment (RedMser) [inline doc/classes/AudioEffectStereoEnhance.xml:17]:
```suggestion
			Widens sound stage through phase shifting in conjuncion with [member time_pullout_ms]. Just pans sound to the left channel if [member time_pullout_ms] is 0.
```

File: doc/classes/AudioEffectStereoEnhance.xml (hunk 11 above)
Comment (betalars) [reply doc/classes/AudioEffectStereoEnhance.xml]: you only caught one of two typos in that word and I just fixed both :D

**Comment by Repiteo** [comment]
Thanks, and congratulations on your first contribution! 🎉
